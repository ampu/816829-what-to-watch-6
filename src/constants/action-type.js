const ActionType = {
  SET_MOVIES_STATUS: `movies/setMoviesStatus`,
  SET_MOVIES: `movies/setMovies`,
  SET_PROMO_STATUS: `movies/setPromoStatus`,
  SET_PROMO_MOVIE: `movies/setPromoMovie`,

  SET_LOGIN_STATUS: `user/setLoginStatus`,
  SET_USER: `user/setUser`,
  SET_MY_LIST_STATUS: `user/setMyListStatus`,
  SET_MY_LIST: `user/setMyList`,

  SET_GENRE: `genres/setGenre`,
};

export default ActionType;
