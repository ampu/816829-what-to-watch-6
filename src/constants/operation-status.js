const OperationStatus = {
  UNSET: `unset`,
  PENDING: `pending`,
  RESOLVED: `resolved`,
  REJECTED: `rejected`,
  UNAUTHORIZED: `unauthorized`,
};

export default OperationStatus;
