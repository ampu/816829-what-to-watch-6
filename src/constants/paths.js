const MainPath = {
  INDEX: `/`,
  SIGN_IN: `/login`,
  MY_LIST: `/mylist`,
  MOVIE: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  PLAYER: `/player/:id`,
  NOT_FOUND: `/not-found`,
};

const MoviePath = {
  MOVIE_OVERVIEW: `/films/:id`,
  MOVIE_DETAILS: `/films/:id/details`,
  MOVIE_REVIEWS: `/films/:id/reviews`,
};

const MOVIE_PATHS = Array.from(new Set([MainPath.MOVIE, ...Object.values(MoviePath)]));

export {
  MainPath,
  MoviePath,
  MOVIE_PATHS,
};
