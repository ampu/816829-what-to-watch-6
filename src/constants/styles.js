const DEFAULT_BACKGROUND_STYLE = {backgroundColor: `black`};
const BORDER_BOTTOM_STYLE = {borderBottomStyle: `rgba(255, 255, 255, 0.24)`};

export {
  DEFAULT_BACKGROUND_STYLE,
  BORDER_BOTTOM_STYLE,
};
