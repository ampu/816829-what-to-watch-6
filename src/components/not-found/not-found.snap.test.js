import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';

import {render} from '@testing-library/react';
import configureStore from 'redux-mock-store';

import Domain from '../../store/domain';
import OperationStatus from '../../constants/operation-status';
import {ALL_GENRES} from '../../constants/genre';

import NotFound from './not-found';

const INITIAL_STATE = {
  [Domain.MOVIE]: {
    moviesStatus: OperationStatus.UNSET,
    movies: [],
    promoStatus: OperationStatus.UNSET,
    promoMovie: undefined,
  },
  [Domain.USER]: {
    loginStatus: OperationStatus.UNSET,
    user: undefined,
  },
  [Domain.GENRE]: {
    genre: ALL_GENRES,
  },
};

const thunkMock = () => (next) => (action) => {
  if (typeof action === `function`) {
    return Promise.resolve();
  }
  return next(action);
};

const initializeStoreWithState = configureStore([thunkMock]);

describe(`NotFound component`, () => {
  it(`Should render according to the snapshot`, () => {
    const {container} = render((
      <redux.Provider store={initializeStoreWithState(INITIAL_STATE)}>
        <Router history={createMemoryHistory()}>
          <NotFound/>
        </Router>
      </redux.Provider>
    ));
    expect(container).toMatchSnapshot();
  });
});
