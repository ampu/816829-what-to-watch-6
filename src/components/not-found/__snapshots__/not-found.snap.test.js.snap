// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component Should render according to the snapshot 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="logo__link logo__link--active"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        Page Not Found
      </h1>
      <div
        class="user-block"
      />
    </header>
    <div
      class="not-found"
    >
      <a
        class="not-found__main-link"
        href="/"
      >
        Go to main page
      </a>
    </div>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="logo__link logo__link--light logo__link--active"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
